xarray/core/dataset.py

```diff
@@ -2384,6 +2384,57 @@ def from_dataframe(cls, dataframe):
             obj[name] = (dims, data)
         return obj
 
+    def to_dask_dataframe(self, set_index=False):
+        """
+        Convert this dataset into a dask.dataframe.DataFrame.
+
+        Non-index variables in this dataset form the columns of the
+        DataFrame.
+
+        If set_index=True, the dask DataFrame is indexed by
+        this dataset's coordinate. Since dask DataFrames to not support
+        multi-indexes, this only works if there is one coordinate dimension.
+        """
+
+        import dask.dataframe as dd
+        import dask.array as da
+
+        columns = [k for k in self if k not in self.dims]
```

Comment:
Why skip variables that are in `self.dims`? I think we want those to be part of the resulting dask dataframe, too -- even though we don't want to set them as an actual index.

Reply:
For multiple dimensions, there wouldn't be a one-to-one correspondence between the coordinate and the data (different sizes).

Reply:
If we call `set_dims()` then all the coordinates also become the same size as the data.

There isn't unique mapping for each coordinate independently, but when you combine them together they unique identify each row (at least assuming the coordinates have unique values to begin with).

More concretely, consider something like the following example:
```
>>> ds = xr.Dataset({'z': (('x', 'y'), [[1, 2], [3, 4]])}, {'x': [1, 2], 'y': ['a', 'b']})

>>> ds
<xarray.Dataset>
Dimensions:  (x: 2, y: 2)
Coordinates:
  * x        (x) int64 1 2
  * y        (y) |S1 'a' 'b'
Data variables:
    z        (x, y) int64 1 2 3 4

>>> ds.to_dataframe()
     z
x y   
1 a  1
  b  2
2 a  3
  b  4
```
Here, `x` and `y` are part of the index, but we would probably want them to be columns.

One interesting case it how to handle dimensions without coordinates. In this case, we probably want to fill in integer columns, again similar to how `to_dataframe()` already works:
```
>>> ds2 = xr.Dataset({'z': (('x', 'y'), [[1, 2], [3, 4]])}, {'y': ['a', 'b']})

>>> ds2
<xarray.Dataset>
Dimensions:  (x: 2, y: 2)
Coordinates:
  * y        (y) |S1 'a' 'b'
Dimensions without coordinates: x
Data variables:
    z        (x, y) int64 1 2 3 4

>>> ds2.to_dataframe()
     z
x y   
0 a  1
  b  2
1 a  3
  b  4
```

Reply:
Done. Coordinate variables are now retained as dask dataframe columns.